### PR TITLE
Documentation of API Console in API Designer

### PR DESCRIPTION
**Pull request by rddheath**
Suggested means of reviewing:
1. Read files in the downloadable site that you can get the .zip for [here](https://drive.google.com/file/d/1SOREcUNYoeq8LejnaS9kE6D2Q5pN-L6d/view?usp=sharing).
2. If you have a comment, copy the filename, click the **Files changed** link above, then search on the filename.
3. Make your comment.

Note: The documentation never mentions the name "API Console" because that name is nowhere in the UI. Nobody is going to search for the documentation of API Console if they are not aware that they are using API Console.
Also, this documentation for API Designer is the first set of documentation for API Console. Subsequent sets are going to be in the documentation for Exchange, APIkit, API Manager, and API Community Manager.

Note 2: This documentation is not complete. I still need to add text to the sections about security schemes and fill out a few other sections. I'm putting it out for review so that reviewers can tell me 1) whether they think it is the right sort of documentation for APIC, and 2) whether they see any mistakes in what I've done so far.

### CODE REVIEW

File: modules/ROOT/pages/design-doc-spec-method.adoc [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+.Display of the documentation for some of the types that are in the body.
+image::apid-doc-get-overview-response-body-types.png[]
+
+*The Numbered Parts of the Image*
+
+. The name of the type.
+. The data type of the type.
+. The description that is defined in the `description` facet for the type.
+. A generated indicator that appears when the specification defines the type as required. You do not need to add the indicator to the definition of the type in a specification.
+. Examples that are defined for the type. If the specification does not define an example and if the media type for examples is defined globally in the specification, then an example is generated. For the type `item.paymentMethodType` in the image, two examples would be generated. The values in generated examples are empty strings.
```
[the remark this answers is not in the document]
Comment (rddheath) [reply modules/ROOT/pages/design-doc-spec-method.adoc]: @harshakirank , the text says "For the type `item.paymentMethodType` in the image, two examples would be generated. The values in generated examples are empty strings." 
The first sentence does _not_ say that the examples in the image is generated; it says that, if item.paymentMethodType was not defined with examples, then its two examples would be generated. 
The second sentence does _not_ say that the values in the example in the image are empty strings. The text says that the values in generated examples are empty strings.

File: modules/ROOT/pages/apid-behavioral-headers.adoc [hunk 2]
```diff
@@ -1,32 +1,37 @@
...
 ifndef::env-site,env-github[]
 include::_attributes.adoc[]
 endif::[]
 
 :imagesdir: ../assets/images
 
-You can use the following behavioral headers to change the behavior of the mocking service. Use behavioral headers where you would normally specify other types of header when simulating requests to API specifications.
+You can use the following headers to change the behavior of the mocking service.
 
+== Introduce Delays into Response Times
```
Comment (runnabro) [inline modules/ROOT/pages/apid-behavioral-headers.adoc]: I think simply:

> Response Time Delays

Is easier to understand.

File: modules/ROOT/pages/apid-behavioral-headers.adoc [hunk 3]
```diff
@@ -1,32 +1,37 @@
...
 You can specify a value for this header as an integer or as a range of integers. If you specify a range, the value must have the form `min-max`. For example, given the value `500-1000`, the request will take a random time between 500 and 1000 milliseconds.
-+
+
 [NOTE]
 ====
 It is possible for the delay to be longer than a single integer value or the high end of a range, depending on the latency of your network.
 ====
 
-MS2-Error-Level:: Indicates which range of error status codes the mocking service is restricted to using. You can specify one of the following three case-sensitive values:
+== Restrict the Range of Error Status Codes That Can Be Returned
```
Comment (runnabro) [inline modules/ROOT/pages/apid-behavioral-headers.adoc]: Similar to above:

> Restrict Status Codes

File: modules/ROOT/pages/apid-behavioral-headers.adoc [hunk 4]
```diff
@@ -1,32 +1,37 @@
...
+
+Use the `MS2-Error-Level` header to indicate which range of error status codes the mocking service is restricted to using. You can specify one of the following three case-sensitive values:
 +
 * `REQUEST_ONLY`: The mocking service uses only 4XX codes.
 * `SERVER_ONLY`: The mocking service uses only 5XX codes.
 * `ALL`: The mocking service can use both 4XX and 5XX codes. This is the default value.
 
-MS2-Error-Rate:: Indicates the probability of getting an error status code as a response. The value for this header must be a decimal between 0 and 1.
-+
+== Randomize the Probability of Receiving Particular Status Codes
```
Comment (runnabro) [inline modules/ROOT/pages/apid-behavioral-headers.adoc]: Similar to above:

> Randomize Status Code Responses

File: modules/ROOT/pages/apid-behavioral-headers.adoc [hunk 5]
```diff
@@ -112,13 +123,17 @@ The possible values of MS2-Status-Code-Selection are:
 +
 `RANDOM`:: The mocking service randomly selects one of the status codes to return. For example, if the specification defines the status codes for a POST method 500, 409, 400, 200, the mocking service returns the response for any one of them.
 
+== Test API Specifications That Contain Errors
```
Comment (runnabro) [inline modules/ROOT/pages/apid-behavioral-headers.adoc]: Less verbose:

> Test API Specifications for Errors

File: modules/ROOT/pages/apid-behavioral-headers.adoc (hunk 2 above)
Comment (rddheath) [reply modules/ROOT/pages/apid-behavioral-headers.adoc]: Each of the headings in this topic tells the user what they can do. So, each heading starts with a verb.

File: modules/ROOT/pages/apid-behavioral-headers.adoc (hunk 3 above)
Comment (rddheath) [reply modules/ROOT/pages/apid-behavioral-headers.adoc]: Same comment as above. Also, your heading is less clear.

File: modules/ROOT/pages/apid-behavioral-headers.adoc (hunk 4 above)
Comment (rddheath) [reply modules/ROOT/pages/apid-behavioral-headers.adoc]: I disagree that your suggestion is more clear.

File: modules/ROOT/pages/apid-behavioral-headers.adoc (hunk 5 above)
Comment (rddheath) [reply modules/ROOT/pages/apid-behavioral-headers.adoc]: No, the section is about testing API specifications _that_ contain errors.

File: modules/ROOT/pages/design-doc-spec-summary.adoc [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+image::apic-spec-summary-page.png[]
+
+*The Numbered Parts of the Image*
+
+. The title that is defined by the `title` node in the specification.
+. The version that is defined by the `version` node.
+. One of three entries:
++
+* If the mocking service is off, this field displays the base URI that is defined by the `baseUri` node in the specification.
+* If the mocking service is on, this field displays the base URI that the mocking service generated and that you can use in tests of the endpoints in the specification.
```
[the remark this answers is not in the document]
Comment (rddheath) [reply modules/ROOT/pages/design-doc-spec-summary.adoc]: @mnbattaglia , clarified.

File: modules/ROOT/pages/design-test-endpoints-code-editor.adoc [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+_Describe where the response appears._
+_Describe how to view the response._
+_Describe that the request URL in the response section includes the proxy info for the mocking service. It is not the same URL as the request._
+
+Example:
+Original request for a GET:
+https://anypoint.mulesoft.com/mocking/api/v1/links/bc88372b-f66e-47f6-a1fc-aba75de27fee/api/customers?firstName=Tim&lastName=Mit
+
+Reported request for the GET:
+https://anypoint.mulesoft.com/apiconsoleproxy/api/v1/proxy/https://anypoint.mulesoft.com/mocking/api/v1/links/bc88372b-f66e-47f6-a1fc-aba75de27fee/api/customers?firstName=Tim&lastName=Mit
```
[the remark this answers is not in the document]
Comment (rddheath) [reply modules/ROOT/pages/design-test-endpoints-code-editor.adoc]: @mnbattaglia , I don't understand what you mean in the first two sentences. Of course, I've heard of CORS, but I don't know what it is and what it does.

**Comment by failup** [comment]
@rddheath We are closing this contribution because it has exceeded the age limit for a timely merge. If the contribution is still valid, please open a separate GitHub issue or pull request.

**Pull request closed without merging**